packages/in_app_purchase/lib/src/store_kit_wrappers/sk_product_wrapper.dart

```diff
@@ -4,9 +4,52 @@
 
 import 'package:flutter/foundation.dart';
 
+/// Response type for a [startProductRequest] request.
+///
+/// Contains information about a list of products and a list of invalid product identifers.
+/// this is a wrapper for StoreKit's [SKProductsResponse](https://developer.apple.com/documentation/storekit/skproductsresponse?language=objc).
+class SkProductResponseWrapper {
+  SkProductResponseWrapper(
+      {@required this.products, @required this.invalidProductIdentifiers});
+
+  /// constructor to build with a map
+  ///
+  /// Used for constructing the class with map passed from the OBJC layer.
+  SkProductResponseWrapper.fromMap(Map<String, List<dynamic>> map)
+      : products = _getListFromMapList(_getProductMapListFromResponseMap(map)),
+        invalidProductIdentifiers =
+            List.castFrom<dynamic, String>(map['invalidProductIdentifiers']);
+
+  /// The list of the products.
+  ///
+  /// Can be null if the [SKProductRequestMaker]'s [startProductRequest] method does not pass correct [productIdentifiers].
```

Comment:
Handling null collections is very error-prone. It would be much better if there was always a `List<>` returned here, but it was just empty in cases where there were no products. That way devs using this won't have to guard against `null` here and can just consistently iterate over the list. (This applies to a few lists in this patch.)